server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -113,6 +116,14 @@ public void setVersion(Version version) {
         this.version = version;
     }
 
+    public boolean hasFeature(final String feature) {
+        return this.features.contains(feature);
+    }
+
+    public void setFeatures(final Set<String> features) {
+        this.features = Collections.unmodifiableSet(new HashSet<>(features));
```

Comment:
assert it's  currently empty?

Reply:
I wrapped this into 26071ff630e5b20be17a4f909416ce2e06691d59.